### ISSUE TITLE
Unable to start go-langserver with Oni 0.2.18

### ISSUE THREAD

**Issue by josemarluedke** [onivim/oni#1061]
Just upgraded to Oni 0.2.18 and now I'm no longer able to get `go-langserver` started. 

It was fine with the previous release. 

<img width="554" alt="screen shot 2017-12-02 at 9 26 23 pm" src="https://user-images.githubusercontent.com/230476/33522700-c00d3f9e-d7a7-11e7-888c-1d320ea9d0ba.png">

**Comment by bryphe** [comment]
Strange - I wonder if there is a change in the `$PATH` behavior due to #934?

Would it be possible to check the following?
- In Oni, run `:term` and then `echo $PATH`
- In a separate bash shell, outside of Oni, run `which go-langserver`

My guess is that Oni is missing some path that is needed to find go-langserver. In the worst case, we can add it to `environment.additionalPaths` as a workaround. Sorry this regressed for you!

**Comment by josemarluedke** [comment]
Inside of Oni:

```
bash-3.2$ echo $PATH
/Users/josemarluedke/.nodenv/shims:/Users/josemarluedke/.rvm/gems/ruby-2.4.2/bin:/Users/josemarluedke/.rvm/gems/ruby-2.4.2@global/bin:/Users/josem
arluedke/.rvm/rubies/ruby-2.4.2/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/josemarluedke/.rvm/bin:/Users/josemarluedke/.rvm/bin:/User
s/josemarluedke/code/go/bin :/usr/bin:/usr/local/bin::/Users/josemarluedke/.fzf/bin:/Users/josemarluedke/.rvm/bin
bash-3.2$
```
```
bash-3.2$ which go-langserver
bash-3.2$
```

Outside of Oni:


```
$ echo $PATH
/Users/josemarluedke/.nodenv/shims:/Users/josemarluedke/.rvm/gems/ruby-2.4.2/bin:/Users/josemarluedke/.rvm/gems/ruby-2.4.2@global/bin:/Users/josemarluedke/.rvm/rubies/ruby-2.4.2/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/josemarluedke/.rvm/bin:/Users/josemarluedke/.rvm/bin:/Users/josemarluedke/code/go/bin
```

```
$ which go-langserver
/Users/josemarluedke/code/go/bin/go-langserve
```

You are probably right, that PR seems to be the cause of it.

**Comment by Akin909** [comment]
@bryphe @josemarluedke  it seems that with that PR by pulling in the `PATH` variable and concatenating it with the users additional paths there is a trailing space added as well as a duplicate colon. `luedke/code/go/bin :/usr/bin:/usr/local/bin::/` here is the section, I believe this makes the variable invalid so the lang server isn't picked up. I think a possible solution would be to remove the additional variables (which might not be ideal although the variables should ideally be in a users path) another solution is to fix the concatenation on `oni` end but this then leaves the door open to an invalid `oni` path due to potential mix up in the config option or at least causes duplicates, which doesn't invalidate the path but can cause very minimal slowdown on executing commands

**Comment by Akin909** [comment]
Done a bit of debugging and it seems the process which generates the `PATH` variable includes a `\n` and an unnecessary extra separator. which I believe is what is causing the path to be invalid

**Comment by bryphe** [comment]
Thanks for the investigation and fix, @Akin909 ! Looks like PR #1066 takes care of this 👍 - I'll close this out.
